plugins/woocommerce/includes/class-wc-tracker.php

```diff
@@ -498,6 +498,69 @@ private static function get_order_dates() {
 		return array_merge( $min_max, $processing_min_max );
 	}
 
+	/**
+	 * Extract the group key for an associative array of objects which have unique ids in the key.
+	 * A 'group_key' property is introduced in the object.
+	 * For example, two objects with keys like 'WooDataPay ** #123' and 'WooDataPay ** #78' would
+	 * both have a group_key of 'WooDataPay **' after this function call.
+	 *
+	 * @param array  $objects     The array of objects that need to be grouped.
+	 * @param string $default_key The property that will be the default group_key.
+	 * @return array Contains the objects with a group_key property.
+	 */
+	private static function extract_group_key( $objects, $default_key ) {
+		$keys = array_keys( $objects );
+
+		// Sort keys by length and then by characters within the same length keys.
+		usort(
+			$keys,
+			function( $a, $b ) {
+				if ( strlen( $a ) === strlen( $b ) ) {
+					return strcmp( $a, $b );
+				}
+				return ( strlen( $a ) < strlen( $b ) ) ? -1 : 1;
+			}
+		);
+
+		// Look for common tokens in every pair of adjacent keys.
+		$prev = '';
+		foreach ( $keys as $key ) {
+			if ( $prev ) {
+				$comm_tokens = array();
+
+				// Tokenize the current and previous gateway names.
+				$curr_tokens = preg_split( '/[ :,\-_]/ ', $key );
+				$prev_tokens = preg_split( '/[ :,\-_]/ ', $prev );
```

Comment:
```suggestion
				$curr_tokens = preg_split( '/[ :,\-_]+/', $key );
				$prev_tokens = preg_split( '/[ :,\-_]+/', $prev );
```

Might be worth detecting more than one token boundary at a time. For example, if we have something like `PayPalTransaction ID: #12345`, the `:` and the ` ` could be considered one boundary instead of two separate ones.

My test data looks like this:

```
cheque: *123
cheque: #456
cheque  #8675
```

Which results in:

```
  ["gateway_cheque #8675_USD_count"]=>
  string(1) "1"
  ["gateway_cheque #8675_USD_total"]=>
  string(6) "325.49"
  ["gateway_cheque _USD_count"]=>
  int(2)
  ["gateway_cheque _USD_total"]=>
  float(650.98)
```

But with the change above, I get:

```
  ["gateway_cheque_USD_count"]=>
  int(3)
  ["gateway_cheque_USD_total"]=>
  float(976.47)
```

Could also take this further here or later on and `trim` the tokens to ensure that resulting keys are more normalized.

Reply:
Thanks @coreymckrill 
That sounds like a good improvisation. I'm committing the suggestion.